File: visitor/oc/bytedesk-oc/core/model/BDQueueModel.m
//
//  KFDSQueueModel.m
//  bdcore
//
//  Created by 萝卜丝 on 2018/11/24.
//  Copyright © 2018年 Bytedesk.com. All rights reserved.
//

#import "BDQueueModel.h"
#import "BDSettings.h"

@implementation BDQueueModel

-(instancetype)initWithDictionary:(NSDictionary *)dictionary {
    
    self = [super init];
    if (self) {
        
        // TODO: 不再引用服务器端id
        _server_id = [dictionary objectForKey:@"id"];
        _qid = [dictionary objectForKey:@"qid"];
        
        if (![[dictionary objectForKey:@"visitor"] isKindOfClass:[NSNull class]]) {
            _visitor_uid = [[dictionary objectForKey:@"visitor"] objectForKey:@"uid"];
            _nickname = [[dictionary objectForKey:@"visitor"] objectForKey:@"nickname"];
            _avatar = [[dictionary objectForKey:@"visitor"] objectForKey:@"avatar"];
            _visitor_client = [[dictionary objectForKey:@"visitor"] objectForKey:@"client"];
        }
        
//        _agent_uid = [[dictionary objectForKey:@"agent"] objectForKey:@"uid"];
//        _status = [dictionary objectForKey:@"status"];
        
        _current_uid = [BDSettings getUid];
    }
    return self;
}

- (BOOL)isEqual:(id)object {
    BDQueueModel *queue = (BDQueueModel *)object;
    if ([_qid isEqualToString:queue.qid]) {
        return TRUE;
    }
    return FALSE;
}


@end

File: visitor/swift/Sources/bytedesk-swift/Models/BDQueueModel.swift
//
//  File.swift
//  
//
//  Created by 宁金鹏 on 2023/9/9.
//

import Foundation

public class BDQueueModel: Codable, Identifiable, ObservableObject {
    
//    var uuId: NSNumber? // 本地表主键id
//    var serverId: NSNumber? // 服务器端主键
    
    var qid: String? // 唯一数字id，保证唯一性
    var nickname: String? // 访客昵称
    var avatar: String? // 访客头像
    var visitorUid: String? // 访客uid
    var visitorClient: String? // 访客来源客户端
    var agentUid: String? // 客服uid
    var agentClient: String? // 客服接入端
    var threadTid: String? // 所属thread tid
    var workgroupWid: String? // 所属工作组 wid
    var actionedAt: String? // 接入客服时间
    var status: String? // queuing/accepted/leaved
    var currentUid: String? // 当前登录用户Uid
    
    
}
